desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java

```diff
@@ -558,6 +582,33 @@ protected boolean onWindowClosing() {
         return windowClosing;
     }
 
+    private boolean isJobUnsaved(FileObject lastSavedJobFile, AnalysisJobBuilder analysisJobBuilder) {
+        if (lastSavedJobFile == null) {
```

Comment:
here we can add a few more precautionary optimizations. Also check:

lastSavedJobFile.exists()
lastSavedJobFile.length() > 0